### PR TITLE
Add ability to add description to pull payment

### LINKED ISSUE #2625: [UI/UX] Add descriptions to pull payments
### Discussed in https://github.com/btcpayserver/btcpayserver/discussions/2586

<div type='discussions-op-text'>

<sup>Originally posted by **dstrukt** June 20, 2021</sup>
Similar to payment requests (which currently has a description field), context is needed for the pull payment.

I'd like to propose a simple description field be added to editor (can be a simple textarea or our wysiwyg) and the end-user view.

Mockups for reference (one with existing editor ui, and potential new create ui) and the end-user view.

Is there anything I'm missing?

Feedback & thoughts welcome!

Screen Shot 2021-06-20 at 6 58 35 AM
 
Screen Shot 2021-06-20 at 6 41 13 AM</div>

### PR DESCRIPTION
**Pull request by bolatovumar**
Added ability to set a description for a pull payment. Also added Greenfield API support.

Screen Shot 2022-01-31 at 9 22 45 PM
Screen Shot 2022-01-31 at 9 27 24 PM

close #2625

### CODE REVIEW

**Comment by dstrukt** [comment]
Hell yeah!

Thanks for picking this one up.

**Comment by bolatovumar** [comment]
Build is failing because something goes wrong when pulling the Docker image

Capture2

File: BTCPayServer/Views/UIPullPayment/ViewPullPayment.cshtml [hunk 1]
```diff
@@ -121,7 +121,7 @@
                             }
                             @if (!string.IsNullOrEmpty(Model.Description) && Model.Description != "<br>")
```
Comment (dennisreimann) [inline BTCPayServer/Views/UIPullPayment/ViewPullPayment.cshtml:122]: Can we eliminate the `Model.Description != "<br>"` when saving the pull payment?

**Review by dennisreimann — COMMENTED** [review]
Looks good!

File: BTCPayServer/Views/UIPullPayment/ViewPullPayment.cshtml (hunk 1 above)
Comment (bolatovumar) [reply BTCPayServer/Views/UIPullPayment/ViewPullPayment.cshtml]: @dennisreimann done here: 2d9d40bcfc34aee393a5699952ffe51d9433366e

Comment (dennisreimann) [reply BTCPayServer/Views/UIPullPayment/ViewPullPayment.cshtml]: Ok, what I meant is that we should set the `Description` to null in that case when the pp gets saved -- this part is still missing, correct?

Comment (bolatovumar) [reply BTCPayServer/Views/UIPullPayment/ViewPullPayment.cshtml]: @dennisreimann why would we do this for PPs specifically? We don't do this for Payment Requests or for anything else that has a description field. Should we do this everywhere where we have a description field?

Comment (dennisreimann) [reply BTCPayServer/Views/UIPullPayment/ViewPullPayment.cshtml]: I think that'd make sense. /cc @Kukks

**Review by Kukks — APPROVED** [review]

**Pull request merged**